topdown/crypto.go

```diff
@@ -177,23 +181,57 @@ func builtinCryptoX509ParseRSAPrivateKey(_ BuiltinContext, operands []*ast.Term,
 	}
 
 	// get the raw private key
-	rawKey, err := getRSAPrivateKeyFromString(string(input))
+	rawKey, err := getPrivateKeyFromString(string(input))
+	if err != nil {
+		return err
+	}
+
+	if len(rawKey.([]crypto.PrivateKey)) != 0 {
+
+		rsaPrivateKey, err := jwk.New(rawKey.([]crypto.PrivateKey)[0])
+		if err != nil {
+			return err
+		}
+
+		jsonKey, err := json.Marshal(rsaPrivateKey)
+		if err != nil {
+			return err
+		}
+
+		if err := util.UnmarshalJSON(jsonKey, &x); err != nil {
+			return err
+		}
+	}
+
+	value, err := ast.InterfaceToValue(x)
+	if err != nil {
+		return err
+	}
+
+	return iter(ast.NewTerm(value))
+}
+
+func builtinCryptoParsePrivateKeys(_ BuiltinContext, operands []*ast.Term, iter func(*ast.Term) error) error {
+
+	a := operands[0].Value
+	input, err := builtins.StringOperand(a, 1)
 	if err != nil {
 		return err
 	}
 
-	rsaPrivateKey, err := jwk.New(rawKey)
+	// get the raw private key
```

Comment:
Before we run `getPrivateKeyFromString` I think we can check that the input is not `""`, if it is, we can return nil and avoid calling `getPrivateKeyFromString`.